ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/manage/[ingestKeys]/Keys.tsx

```diff
@@ -22,27 +25,47 @@ type KeysProps = {
   environmentSlug: string;
 };
 
-export default async function Keys({ environmentSlug }: KeysProps) {
-  const environment = await getEnvironment({
+const LoadingSkeleton = () => (
+  <div className="border-b border-slate-100 px-4 py-3">
+    <Skeleton className="mb-1 block h-11 w-full" />
+  </div>
+);
+
+export default function Keys({ environmentSlug }: KeysProps) {
+  const [{ data: environment, fetching: fetchingEnvironment }] = useEnvironment({
     environmentSlug,
   });
-
-  const response = await graphqlAPI.request(GetKeysDocument, {
-    environmentID: environment.id,
+  const [{ data, fetching: fetchingKey }] = useQuery({
+    query: GetKeysDocument,
+    variables: {
+      environmentID: environment?.id!,
```

Comment:
We shouldn't use the `!` operator because it's saying "trust me, bro" to the type system, which often leads to bugs. It's the same risk as type casting with `as`